### DISCUSSION TITLE
SAMx7x DMA

### DISCUSSION THREAD

**Discussion by lgili** [modm-io/modm#978] [General]
I don't know if here is the right place to ask, but there are someone working on DMA for samx7x devices?

I need this, and I'm thinking to implemented that. There are some advises or tips to how do that?

I appreciate any input about that.

**Comment by salkinium** [comment]
Well, modm-devices already has [DMA data for the SAMx7x series](https://github.com/modm-io/modm-devices/pull/96), so I assume there was an intention to add a driver for it too. Perhaps there's already code on a fork somewhere? @rleh @chris-durand?

I'm not at all familiar with the SAM hardware, so I don't know how difficult it is to add. In the past we first implemented a driver for only a single device (by returning `True` from the `module.lb:prepare()` function for only one `options[":target"]`), so that you don't have to make a generic driver.

**Comment by lgili** [comment]
Yes, I meant to add a driver for DMA.

The idea to start with just one device is a good one. It will facilitate for me to get started.
I already have some code for same70 DMA, maybe I can do this if I start with just one device.

**Comment by chris-durand** [comment]
I've also been working on an implementation but I have completely missed this discussion while I was on vacation. You find it in #987 and I expect to finish it soon because I have deadlines at work :smile:. I hope it will be useful for you as well.

**Comment by lgili** [comment]
I have my DMA implementation running for a while in one of our products. I have it integrated with ADC and I2C and it's working fine. Now I'm testing the modm and I've created a test [repo](https://github.com/lgili/modm-template/tree/develop) and I'm porting my code to momd style. And I will look at your code which will certainly be of great help, thank you very much. If you want to take a look at my code feel free. And if you need some help or have any questions I'm happy to help, I've been working with SAM chips for a while.
